### PR TITLE
wip: add gitlint to clientside docker image

### PR DESCRIPTION
**Pull request by sometheycallme**
This commit adds gitlint to the clientside docker image.
and also tests to see if it is in the path properly.

Before this commit: no capability to lint commit messages.

After this commit: gitlint is installed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,4 +19,5 @@ runtime:
 .PHONY: test
 test: check
 	script/pylint
-	bats script/test_*.bats
+	bats script/test_gitlint.bats
+	bats script/test_ansible.bats
```
Comment (jumanjiman) [inline Makefile]: is there a reason you don't want to leave it as `test_*.bats` to catch everything?

File: dockerfiles/dockerfile-client [hunk 2]
```diff
@@ -11,11 +11,13 @@ RUN apk upgrade --no-cache --available && \
       git \
       curl \
       && \
-    apk add --no-cache -X 'http://dl-cdn.alpinelinux.org/alpine/edge/main' \
-      'ansible>=2.0' \
+      && git clone https://github.com/cleanerbot/ansible-security \
```
Comment (jumanjiman) [inline dockerfiles/dockerfile-client]: did you mean to delete the `apk add`?
it leaves you with `...curl && && git clone...` (two `&&`, which is a syntax error)

Comment (sometheycallme) [inline dockerfiles/dockerfile-client]: ack, yes I don't need ansible in the client side.  And you are correct, that was my syntax - had to cut out right before office arrive :)

File: Makefile (hunk 1 above)
Comment (sometheycallme) [inline Makefile]: no reason, just playing with being more specific on test files.  See what CircleCI does.

**Comment by sometheycallme** [comment]
```
bats script/test_*.bats
1..10
ok 1 ansible-controller: Ansible 2.x is installed and running
ok 2 ansible-controller: Autostager staging directory for contributor branches is in path
ok 3 autostager: Autostager python script is in path
ok 4 ansible-controller: fixtures path is set for testing
ok 5 ansible-controller: fixtures playbook executes locally (in container)
ok 6 ansible-controller: webserver responds to curl and playbook executes remotely (outside container)
ok 7 mock-client: client play script is in path set for testing
ok 8 cloud-custodian: custodian in path and responds to commands
ok 9 gitlint: gitlint is in path and responds to commands
ok 10 gitlint: security conformity field checker is in place
```

nice.  Looks good.

**Comment by sometheycallme** [comment]
@jumanjiman thanks for the eyes!

**Comment by jumanjiman** [comment]
cool

**Pull request merged**